examples/chassis/typeC.cc

```diff
@@ -0,0 +1,77 @@
+*  Copyright (C) 2024 RoboMaster.                                          *
+*  Illini RoboMaster @ University of Illinois at Urbana-Champaign          *
+*                                                                          *
+*  This program is free software: you can redistribute it and/or modify    *
+*  it under the terms of the GNU General Public License as published by    *
+*  the Free Software Foundation, either version 3 of the License, or       *
+*  (at your option) any later version.                                     *
+*                                                                          *
+*  This program is distributed in the hope that it will be useful,         *
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+*  GNU General Public License for more details.                            *
+*                                                                          *
+*  You should have received a copy of the GNU General Public License       *
+*  along with this program. If not, see <http://www.gnu.org/licenses/>.    *
+*                                                                          *
+****************************************************************************/
+
+#include "main.h"
+#include "bsp_print.h"
+#include "chassis.h"
+#include "cmsis_os.h"
+#include "dbus.h"
+
+bsp::CAN* can = nullptr;
+control::MotorCANBase* fl_motor = nullptr;
+control::MotorCANBase* fr_motor = nullptr;
+control::MotorCANBase* bl_motor = nullptr;
+control::MotorCANBase* br_motor = nullptr;
+
+control::Chassis* chassis = nullptr;
+remote::DBUS* dbus = nullptr;
+
+void RM_RTOS_Init() {
+  print_use_uart(&huart1);
+
+  can = new bsp::CAN(&hcan1);
+  fl_motor = new control::Motor3508(can, 0x201);
+  fr_motor = new control::Motor3508(can, 0x204);
+  bl_motor = new control::Motor3508(can, 0x202);
+  br_motor = new control::Motor3508(can, 0x203);
+
+  control::MotorCANBase* motors[control::FourWheel::motor_num];
+  motors[control::FourWheel::front_left] = fl_motor;
+  motors[control::FourWheel::front_right] = fr_motor;
+  motors[control::FourWheel::back_left] = bl_motor;
+  motors[control::FourWheel::back_right] = br_motor;
+
+  control::chassis_t chassis_data;
+  chassis_data.motors = motors;
+  chassis_data.model = control::CHASSIS_MECANUM_WHEEL;
+  chassis = new control::Chassis(chassis_data);
+
+  dbus = new remote::DBUS(&huart3);
+}
+
+void RM_RTOS_Default_Task(const void* args) {
+  UNUSED(args);
+  osDelay(500);  // DBUS initialization needs time
+
+  control::MotorCANBase* motors[] = {fl_motor, fr_motor, bl_motor, br_motor};
+
+  while (true) {
+    chassis->SetSpeed(dbus->ch0, dbus->ch1, dbus->ch2);
+
+   // Kill switch
+    if (dbus->swr == remote::UP || dbus->swr == remote::DOWN) {
+     RM_ASSERT_TRUE(false, "Operation killed");
+    }
+
+    chassis->Update(false, 30, 20, 60);
```

Comment:
nit: Consider naming arguments, or comment usage? Same for SetSpeed above.
// Update(bool power_limit_on, float power_limit, float chassis_power, float chassis_power_buffer)